lib/parser/provenance.cc

```diff
@@ -328,6 +360,45 @@ std::optional<ProvenanceRange> CookedSource::GetProvenanceRange(
   return {ProvenanceRange{first.start(), last.start() - first.start()}};
 }
 
+std::optional<CharBlock> CookedSource::GetCharBlockFromLineAndColumns(
+    int line, int column1, int column2) const {
+  // 2nd column is exclusive, meaning it is target column + 1.
+  auto provenanceStart{allSources_.GetFirstFileProvenance().value().start()};
```

Comment:
`GetFirstFileProvenance()` could return `std::nullopt`

Reply:
I figured we would never come across a situation where we don't find a file while running this function so I didn't add a guard. Maybe it's a good idea to guard it anyways. What do you think?

Reply:
I think you're okay. Unless someone else can think of a case where it could be a problem.